beacon_node/http_api/src/lib.rs

```diff
@@ -2467,11 +2466,13 @@ pub fn serve<T: BeaconChainTypes>(
         .and(warp::path::end())
         .and(warp::header::optional::<api_types::Accept>("accept"))
         .then(
-            |chain: Arc<BeaconChain<T>>,
+            |light_client_server_enabled: Result<(), Rejection>,
+             chain: Arc<BeaconChain<T>>,
              task_spawner: TaskSpawner<T::EthSpec>,
              block_root: Hash256,
              accept_header: Option<api_types::Accept>| {
                 task_spawner.blocking_response_task(Priority::P1, move || {
+                    light_client_server_enabled?;
```

Comment:
Any reason why not to place this above the task spawner so we avoid spawning a blocking task just to return the error? I noticed we're also doing the same thing on other endpoints.

Reply:
We need to return a future here, so we'd have to double nest `async move`, which is probably fine.

But also, I don't think we should care about performance for endpoints that users shouldn't be calling anyway. The common case is that you might call this once, it fails, and then you restart the node with the light client server enabled